packages/web-components/src/player-component/player.class.ts

```diff
@@ -679,11 +681,47 @@ export class PlayerWrapper {
         return this.getClockTimeString(time, false);
     }
 
-    private onErrorEvent(event: shaka_player.PlayerEvents.ErrorEvent) {
+    private async onErrorEvent(event: shaka_player.PlayerEvents.ErrorEvent) {
         // Extract the shaka.util.Error object from the event.
         // eslint-disable-next-line no-console
         Logger.log(event.detail);
-        if (this.errorCallback) {
+
+        // If we encounter a video error, reset video source UNLESS we exceed
+        // max video errors within three minutes.
+        let errorHandled = false;
+        if (event.detail.code === 3016) {
```

Comment:
Instead of 3016 it should be shaka.util.Error.Code.VIDEO_ERROR .

Reply:
Thanks for the comment Prakash, will incorporate your suggestion in a new PR and abandon this one.